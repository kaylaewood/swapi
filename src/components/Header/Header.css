header {
  display: flex;
  align-items: center;
  padding-top: 5vw;
}

.header-black-box {
  background-color: #0B1E36;
  color: #0B1E36;
  width: 3%;
  height: 4.2vw;
  box-shadow: 15px 15px 15px #A6A6A6;
}

.death-star-container {
  background-color: #0B1E36;
  border-radius: 100%;
  height: 10vw;
  width: 10vw;
  display: flex;
  justify-content: start;
  align-items: center;
  margin-left: -1vw;
  box-shadow: 15px 15px 15px #A6A6A6;
}

img {
  width: 85%;
  margin-left: 5%;
}

h1 {
  background-color: #0B1E36;
  color: white;
  width: 90%;
  text-align: left;
  padding: .3vw;
  padding-left: 1vw;
  margin-left: -1vw;
  font-size: 3vw;
  letter-spacing: .3vw;
  box-shadow: 15px 15px 15px #A6A6A6;
}
