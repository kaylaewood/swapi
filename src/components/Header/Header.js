import React from 'react';
import deathStar from '../../images/death-star.png';
import './Header.css';
import { Link } from 'react-router-dom';

function Header({ heading, showHomeButton, logOut }) {
  return (
    <header>
      <div className='header-black-box'>x</div>
      <div className='death-star-container'>
        <img src={deathStar}/>
      </div>
      <div className='heading-line'>
        <h1>{heading}</h1>
        {
          showHomeButton ?
          <Link to='/'><button className='logout-button' onClick={() => logOut()}>LOG OUT</button></Link> :
          <></>
        }
      </div>
    </header>
  );
}

export default Header;
