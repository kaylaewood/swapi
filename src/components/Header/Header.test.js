import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();
    })
})
