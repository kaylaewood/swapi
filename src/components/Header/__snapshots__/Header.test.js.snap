// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header>
  <div
    className="header-black-box"
  >
    x
  </div>
  <div
    className="death-star-container"
  >
    <img
      src="death-star.png"
    />
  </div>
  <div
    className="heading-line"
  >
    <h1 />
  </div>
</header>
`;
